pkg/plugin/plugin.go

```diff
@@ -18,9 +19,68 @@ const (
 	MethodSendNotification = "SendNotification"
 )
 
+// ConfigOption describes a config element of the channel form
+type ConfigOption struct {
+	// Element name
+	Name string `json:"name"`
+
+	// Element type:
+	//
+	//  string = text, number = number, bool = checkbox, text = textarea, option = select, options = select[multiple], secret = password
+	Type string `json:"type"`
+
+	// Element label map. Locale in the standard format (language_REGION) as key and corresponding label as value.
+	// Locale is assumed to be UTF-8 encoded (Without the suffix in the locale)
+	//
+	//  e.g. {"en_US": "Save", "de_DE": "Speichern"}
+	//  An "en_US" locale must be given as a fallback
+	Label map[string]string `json:"label"`
+
+	// Element placeholder
+	Placeholder string `json:"placeholder,omitempty"`
+
+	// Element title: When the user moves the mouse pointer over an element, a tooltip is displayed with a given message.
+	Help map[string]string `json:"help,omitempty"`
+
+	// Element default value: bool for checkbox, string for other elements
+	Default any `json:"default,omitempty"`
+
+	// Set true if this element is required, omit otherwise
+	Required bool `json:"required,omitempty"`
+
+	// Options of a select element: key => value.
+	// Only required for the type option or options
+	//
+	//  e.g., map[string]string{
+	//			"1":   "January",
+	//			"2":  "February",
+	//		}
+	Options map[string]string `json:"options,omitempty"`
+
+	// Element's min option defines the minimum allowed number value. It can only be used for the type number.
+	Min types.Int `json:"min,omitempty"`
+
+	// Element's max option defines the maximum allowed number value. It can only be used for the type number.
+	Max types.Int `json:"max,omitempty"`
+}
+
+// Info contains plugin information.
 type Info struct {
-	Name             string          `json:"display_name"`
-	ConfigAttributes json.RawMessage `json:"config_attrs"`
+	Type             string          `db:"type"`
```

Comment:
That field should be explicitly tagged with `json:"-"` so that it isn't exposed there at all.